### ISSUE TITLE
Change monetary amounts to BigDecimal

### ISSUE THREAD

**Issue by arkhadi** [mindee/mindee-api-java#151]
### Prerequisites

Put an X between the brackets on this line if you have done all the following:

* [X ] Reproduced the problem or exposed a new need
* [X ] Checked the GitHub existing issues

### Description

When working with monetary amounts in Java it is recommended to avoid the use of Long or Double as they may introduce rounding issues. BigDecimal is the recommended type for monetary amounts. Explanation can be found [here](https://adevait.com/java/java-data-types)

### Steps to Reproduce

**Expected behavior:**

Amount values parsing to BigDecimal from Json.

**Actual behavior:**

Double being used.

**Reproduces how often:**

Every time as it is in all the classes.

### Versions

Latest

**Comment by ianardee** [comment]
Yes, this is something we plan on doing but would be a major breaking change for all users.

There are also some breaking changes we want to do around error handling, we'll do both of these changes in a version 5.0.

**Comment by arkhadi** [comment]
Glad to hear and thanks for the quick reply. I have already implemented the change for the types I'm using so I'm covered.
